Allow confined sysadmin to use tool vipw and vigr

Koncpa opened the pull request:
**Commit 1:**
Allow confined sysadmin to use vipw and vigr, which edits passwd,
shadow, group, gshadow.. Dontaudit manage files and dir labeled
with admin_home_t. Also vipw need to use sss_cache tool.
Allow domain transition from sysadm_passwd_t to sssd_exec_t.

**Commit 2:**
Vigr mechanism of editing group and passwd
files work on principle of recreating the current
file with new changes. Due to this mechanism is
need to again relabel file with selinux label.
Creating interface allowing relabel to the passwd_file_t
type. Allow relabeling for sysadm_passwd_t domain.

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2049018

Koncpa committed:
Allow confined sysadmin to use tool vipw

Allow confined sysadmin to use vipw and vigr, which edits passwd,
shadow, group, gshadow.. Dontaudit manage files and dir labeled
with admin_home_t. Also vipw need to use sss_cache tool.
Allow domain transition from sysadm_passwd_t to sssd_exec_t.

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2049018

zpytela commented:
@Koncpa have you also tried `vipw -s` and `vigr`? They are the same binary, but need access to different resources.

Koncpa commented:
@zpytela  I have tested only vipw -s but not vigr, I can try the vigr tool and I'll let you know.

zpytela commented:
Note they are used to edit passwd, shadow, group, gshadow.

Koncpa commented:
@zpytela I have tried to use `vigr` tool on machine which contains this proposed fix. It appeared denials there related with managing `admin_home_t` dir. So I added rules which allowing this denial to this proposed fix.

zpytela commented:
I am not sure if it is worth just because the editor wants to update its config files, another editor could use a different path. You should rather try to modify the files as this is what the tool is for:
```
-
type=PROCTITLE msg=audit(02/09/2022 10:54:23.864:1085) : proctitle=/usr/bin/vim /etc/group.edit 
type=PATH msg=audit(02/09/2022 10:54:23.864:1085) : item=0 name=/etc/group.edit inode=77407 dev=00:1f mode=file,644 ouid=root ogid=root rdev=00:00 obj=staff_u:object_r:shadow_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0 
type=CWD msg=audit(02/09/2022 10:54:23.864:1085) : cwd=/root 
type=SYSCALL msg=audit(02/09/2022 10:54:23.864:1085) : arch=x86_64 syscall=setxattr success=no exit=EACCES(Permission denied) a0=0x560a0633bcf0 a1=0x7fef45386251 a2=0x560a063567e0 a3=0x23 items=1 ppid=88149 pid=88150 auid=staff uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=pts1 ses=7 comm=vim exe=/usr/bin/vim subj=staff_u:sysadm_r:sysadm_passwd_t:s0-s0:c0.c1023 key=(null) 
type=AVC msg=audit(02/09/2022 10:54:23.864:1085) : avc:  denied  { relabelto } for  pid=88150 comm=vim name=group.edit dev="vda2" ino=77407 scontext=staff_u:sysadm_r:sysadm_passwd_t:s0-s0:c0.c1023 tcontext=system_u:object_r:passwd_file_t:s0 tclass=file permissive=0 
```
```
f35# sesearch -A -s sysadm_passwd_t -c file -p relabelto,relabelfrom
allow sysadm_passwd_t etc_t:file { relabelfrom relabelto };
allow sysadm_passwd_t shadow_t:file { append create getattr ioctl link lock open read relabelfrom relabelto rename setattr unlink watch watch_reads write };
```

Koncpa committed:
Allow sysadm_passwd_t to relabel passwd and group files

Vigr mechanism of editing group and passwd
files work on principle of recreating the current
file with new changes. Due to this mechanism is
need to again relabel file with selinux label.
Creating interface allowing relabel to the passwd_file_t
type. Allow relabeling for sysadm_passwd_t domain.
Allow dac_override permission for sysadm_passwd_t type.

zpytela commented on policy/modules/admin/usermanage.te:
```diff
@@ -500,6 +501,8 @@ init_dontaudit_rw_utmp(sysadm_passwd_t)
 
 logging_send_syslog_msg(sysadm_passwd_t)
 
+userdom_manage_admin_dirs(sysadm_passwd_t)
```
Thinkig about it again, I'd rather not have it here just because an editor wants to update a config file.

zpytela commented on policy/modules/system/authlogin.if:
```diff
@@ -849,6 +849,26 @@ interface(`auth_relabel_shadow',`
 
+#######################################
+## <summary>
+##      Relabel to the
```
Please use tabs to indent.

zpytela reviewed (changes requested):
Otherwise looks fine.

zpytela commented on policy/modules/admin/usermanage.te:
```diff
@@ -500,6 +501,7 @@ init_dontaudit_rw_utmp(sysadm_passwd_t)
 
 logging_send_syslog_msg(sysadm_passwd_t)
 
+userdom_manage_admin_files(sysadm_passwd_t)
```
This one should IMHO go away too.

Koncpa replied:
Do you know about other way how to allow access? I have created file transition from admin_home_t to passwd_file_t and shadow_t, but AVC's messages are still appears.

Koncpa replied:
@zpytela

zpytela replied:
I am okay with vipw not writing to .viminfo.

zpytela commented on policy/modules/system/authlogin.if:
```diff
@@ -849,6 +849,26 @@ interface(`auth_relabel_shadow',`
+## </param>
+#
+interface(`auth_relabelto_passwd_files',`
+        gen_require(`
```
Especially here it is important to indent with tabs.

zpytela reviewed (commented):
So the dac_override is not needed in the end?
